src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -7698,11 +7679,7 @@ internal void WriteSqlDecimal(SqlDecimal d, TdsParserStateObject stateObj)
                 stateObj.WriteByte(0);
 
             Span<uint> data = stackalloc uint[4];
-#if NET8_0_OR_GREATER
             d.WriteTdsValue(data);
-#else
-            SqlTypeWorkarounds.SqlDecimalExtractData(d, out data[0], out data[1], out data[2], out data[3]);
```

Comment:
Can the type workarounds be removed  now?

Reply:
A quick look says at least some of it can be removed. I'll leave it to @benrr101 because the type workarounds are split across netcore and netfx, so they'll likely be touched by the merge efforts.